### ISSUE TITLE
compose up in multi-root workspaces

### ISSUE THREAD

**Issue by hi0t** [microsoft/vscode-docker#3058]
Repro steps:
1.  create a workspace with two directors having a configuration:
```JSON
{
    "folders": [
        {
            "path": "docker"
        },
        {
            "path": "src"
        },
    ],
}
```
2. add docker-compose.yml to the docker directory;
3. invoke command "compose up" by clicking on the file.

We see a drop-down menu with a choice of workspace:
![image](https://user-images.githubusercontent.com/27971344/125517111-92cd20f1-21d6-452e-ac23-28fc6be6cbe8.png)
I clicked on a specific file, why this choice?

**Comment by karolz-ms** [comment]
The prompt helps determine the context for variable substitution.  https://code.visualstudio.com/docs/containers/reference#_command-customization https://code.visualstudio.com/docs/editor/variables-reference 

Have you considered using a custom task? https://code.visualstudio.com/docs/containers/reference#_docker-compose-task That would avoid the prompt.

**Comment by hi0t** [comment]
Thanks for the advice. It works. But in my opinion, this is not logical. I launch "compos up" from the file context menu, then re-select workspace.

**Comment by karolz-ms** [comment]
Agree it is unfortunate. Perhaps we can implement a heuristics for multi-root workspaces, like, use the root that contains the Compose file.

**Comment by bwateratmsft** [comment]
I think I have an idea what's going on here--we do the `quickPickWorkspaceFolder` command even though the compose file URI could probably be used to determine the workspace folder automatically.

**Comment by bwateratmsft** [comment]
We can use the `getWorkspaceFolder` function in the [workspace](https://code.visualstudio.com/api/references/vscode-api#workspace) namespace.

**Comment by bwateratmsft** [comment]
A fix for this has been released in Docker extension version 1.16.0.
